jwst/outlier_detection/utils.py

```diff
@@ -115,14 +92,14 @@ def median_without_resampling(input_models,
             if i == 0:
                 input_shape = (ngroups,)+drizzled_model.data.shape
                 dtype = drizzled_model.data.dtype
-                median_computer = _make_median_computer(input_shape, in_memory, buffer_size, dtype)
+                computer = MedianComputer(input_shape, in_memory, buffer_size, dtype)
 
             weight_threshold = compute_weight_threshold(drizzled_model.wht, maskpt)
             drizzled_model.data[drizzled_model.wht < weight_threshold] = np.nan
-            _append_to_median_computer(median_computer, i, drizzled_model.data, in_memory)
+            computer.append(drizzled_model.data, i)
```

Comment:
Somewhat unrelated to this PR (so feel free to say "make a ticket") but doesn't the above line modify the input model data?

Reply:
I think you mean a different line.  How does appending `drizzled_model.data` to the computer modify it?

I'm a little confused about the "never modify the input data" rule.  Does that mean that for a step like outlier detection where you have a ModelLibrary input and output, you should never call `modify=True` when shelving?

I'd say let's put any changes into a different refactor or PR, unless you have concerns w.r.t. the functions that are going into stcal.

Reply:
Line 98 is that one that can modify the input data. I opened an issue: https://github.com/spacetelescope/jwst/issues/8845

The "never modify the input data" rule is certainly beyond the scope of this PR (and doesn't really apply for most if not all of image 3).

No concerns about this PR. I just brought it up since I noticed it while working on the corresponding romancal PR and wanted to check if I was mistaken.